plugins/wasmedge_opencvmini/opencvmini_func.cpp

```diff
@@ -67,6 +70,101 @@ Expect<uint32_t> WasmEdgeOpenCVMiniBlur::body(const Runtime::CallingFrame &,
   return Env.insertMat(Dst);
 }
 
+Expect<uint32_t>
+WasmEdgeOpenCVMiniBilateralFilter::body(const Runtime::CallingFrame &,
+                                        uint32_t SrcMatKey, uint32_t D,
+                                        double SigmaColor, double SigmaSpace) {
+  cv::Mat Dst;
+  if (auto Src = Env.getMat(SrcMatKey); Src) {
+    cv::bilateralFilter(*Src, Dst, D, SigmaColor, SigmaSpace);
+  }
+  return Env.insertMat(Dst);
+}
+
+Expect<uint32_t>
+WasmEdgeOpenCVMiniBoxFilter::body(const Runtime::CallingFrame &,
+                                  uint32_t SrcMatKey, uint32_t Ddepth,
+                                  uint32_t KernelWidth, uint32_t KernelHeight) {
+  cv::Mat Dst;
+  if (auto Src = Env.getMat(SrcMatKey); Src) {
+    cv::boxFilter(*Src, Dst, Ddepth, cv::Size(KernelWidth, KernelHeight));
+  }
+  return Env.insertMat(Dst);
+}
+
+Expect<uint32_t> WasmEdgeOpenCVMiniDilate::body(const Runtime::CallingFrame &,
+                                                uint32_t SrcMatKey,
+                                                cv::InputArray Kernel) {
```

Comment:
I don't think this will work, the caller of this function is **wasm instance**, it cannot attach **host**'s(e.g. WasmEdge) memory. Thus, passing array usually requires we send a pair of `address x length`.

The problem can be more annoying here when you passing complicated object like `cv::InputArray`, you simply have no good way to pass it to **host**, that's also why we use `SrcMatKey` instead of `cv::Mat`.

You can follow the pattern and use `KernelMatKey`, by adding a `WasmEdgeOpenCVMiniEmptyMat : () -> MatKey` function.

Reply:
@dannypsnl I have added WasmEdgeOpenCVMiniEmptyMat and it returns the flag of cv::Mat(which I believe you have used as key). Is this sufficient or should I add something more?

Reply:
Yes, it’s sufficient